### PR TITLE
MULE-14990: null values causes Exception in parameterNeedsTransformation

### PR DESCRIPTION
**Pull request by ndinu**

### CODE REVIEW

File: src/test/java/org/mule/extensions/internal/execution/JavaArgumentTransformationTestCase.java [hunk 1]
```diff
@@ -118,7 +119,23 @@ public void invokeStaticWithListOfMapsAndStringsTransformed() throws Exception {
...
+  }
 
+  @Test
+  public void invokeStaticWithMapWithListValuesAndOneIsNull() throws Exception {
+    TypedValue<String> payload = flowRunner("invokeStaticWithMapWithListValuesAndOneIsNull")
+        .run()
+        .getMessage()
+        .getPayload();
+    assertThat(payload.getValue(), is("11"));
+  }
```
Comment (alegmarra) [inline src/test/java/org/mule/extensions/internal/execution/JavaArgumentTransformationTestCase.java]: add one negative test where the parameter does not accept a null value and invocation fails
